internal/validation/validation_test.go

```diff
@@ -317,3 +319,113 @@ func TestValidatePending(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateBlock(t *testing.T) {
+	baseBlk := block.Block{
+		Version:        1,
+		Height:         0,
+		PreviousHash:   hashing.New([]byte{'x'}),
+		MerkleRootHash: hashing.New([]byte{'q'}),
+		Timestamp:      time.Now().UnixNano(),
+		Data:           [][]byte{hashing.New([]byte{'r'})},
+	}
+	baseBlk.DataLen = uint16(len(baseBlk.Data))
+	tests := []struct {
+		name  string
+		block block.Block
+		want  bool
+	}{
+		{
+			"Valid Block",
+			baseBlk,
+			true,
+		},
+		{
+			"Invalid version",
+			block.Block{
+				Version:        10000,
+				Height:         baseBlk.Height,
+				PreviousHash:   baseBlk.PreviousHash,
+				MerkleRootHash: baseBlk.MerkleRootHash,
+				Timestamp:      baseBlk.Timestamp,
+				Data:           baseBlk.Data,
+				DataLen:        baseBlk.DataLen,
+			},
+			false,
+		},
+		{
+			"Invalid height",
+			block.Block{
+				Version:        baseBlk.Version,
+				Height:         999999999,
+				PreviousHash:   baseBlk.PreviousHash,
+				MerkleRootHash: baseBlk.MerkleRootHash,
+				Timestamp:      baseBlk.Timestamp,
+				Data:           baseBlk.Data,
+				DataLen:        baseBlk.DataLen,
+			},
+			false,
+		},
+		{
+			"Invalid previous hash",
+			block.Block{
+				Version:        baseBlk.Version,
+				Height:         baseBlk.Height,
+				PreviousHash:   hashing.New([]byte{'a'}),
+				MerkleRootHash: baseBlk.MerkleRootHash,
+				Timestamp:      baseBlk.Timestamp,
+				Data:           baseBlk.Data,
+				DataLen:        baseBlk.DataLen,
+			},
+			false,
+		},
+		{
+			"Invalid previous hash (nil)",
+			block.Block{
+				Version:        baseBlk.Version,
+				Height:         baseBlk.Height,
+				PreviousHash:   nil,
+				MerkleRootHash: baseBlk.MerkleRootHash,
+				Timestamp:      baseBlk.Timestamp,
+				Data:           baseBlk.Data,
+				DataLen:        baseBlk.DataLen,
+			},
+			false,
+		},
+		{
+			"Invalid timestamp",
+			block.Block{
+				Version:        baseBlk.Version,
+				Height:         baseBlk.Height,
+				PreviousHash:   baseBlk.PreviousHash,
+				MerkleRootHash: baseBlk.MerkleRootHash,
+				Timestamp:      baseBlk.Timestamp - 1,
+				Data:           baseBlk.Data,
+				DataLen:        baseBlk.DataLen,
+			},
+			false,
+		},
+		{
+			"Invalid timestamp (future)",
+			block.Block{
+				Version:        baseBlk.Version,
+				Height:         baseBlk.Height,
+				PreviousHash:   baseBlk.PreviousHash,
+				MerkleRootHash: baseBlk.MerkleRootHash,
+				Timestamp:      baseBlk.Timestamp + 1,
```

Comment:
This can be a valid timestamp, it just can't be in the future from "now", where "now" is the point at which the validation is called.